### PR TITLE
#982 Summary emails

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
Commit by commit.

- Add index, isFirst, isLast to summary renderer which is needed for emails as gaps cannot be resolved by flex + gap, cleanup
- Install react-email
- Create summary email
- Create generatePageScreenshot test utility
- Create tests for renderSummaryEmail
- Generate snapshots for emails

### CODE REVIEW

File: forms-shared/src/summary-email/renderSummaryEmail.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const summaryJson = getSummaryJsonNode(
+    formDefinition.schemas.schema,
+    formDefinition.schemas.uiSchema,
+    formData,
+  )
+  const fileInfos = mergeClientAndServerFilesSummary([], serverFiles)
+  const validatedSummary = validateSummary(formDefinition.schemas.schema, formData, fileInfos)
+
+  return render(
+    <SummaryEmail summaryJson={summaryJson} validatedSummary={validatedSummary}></SummaryEmail>,
```
Comment (radoslavzeman) [inline forms-shared/src/summary-email/renderSummaryEmail.tsx:31]: no children -> `<SummaryEmail summaryJson={summaryJson} validatedSummary={validatedSummary} />`

File: forms-shared/tests/summary-email/__image_snapshots__/render-summary-email-ts-render-summary-email-komunitne-zahrady-example-summary-email-should-match-mobile-screenshot-snapshot-1-snap.png
Comment (radoslavzeman) [inline forms-shared/tests/summary-email/__image_snapshots__/render-summary-email-ts-render-summary-email-komunitne-zahrady-example-summary-email-should-match-mobile-screenshot-snapshot-1-snap.png:1]: Quite tight spacing. I would use padding 20px if possible and if it makes sense.

**Review by radoslavzeman — COMMENTED** [review]
Would it be possible to use tailwind for styling also in `forms-shared`?

**Review by radoslavzeman — APPROVED** [review]

File: forms-shared/tests/summary-email/__image_snapshots__/render-summary-email-ts-render-summary-email-komunitne-zahrady-example-summary-email-should-match-mobile-screenshot-snapshot-1-snap.png
Comment (MarekBodingerBA) [reply forms-shared/tests/summary-email/__image_snapshots__/render-summary-email-ts-render-summary-email-komunitne-zahrady-example-summary-email-should-match-mobile-screenshot-snapshot-1-snap.png:1]: Where exactly?

**Comment by MarekBodingerBA** [comment]
> Would it be possible to use tailwind for styling also in `forms-shared`?

Yes, but https://react.email/docs/components/tailwind#known-limitations + https://react.email/docs/components/tailwind#tailwind-configuration-with-px-instead-of-rem. The project already contains Tailwind but a newer version that is not supported by react-email. 

Also, there are many issues with using Tailwind - https://github.com/resend/react-email/issues?q=is%3Aissue+is%3Aopen+tailwind

**Pull request merged**